rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
I'm even more confused now. There must be a misunderstanding somewhere...

WRT by clauses in other cases: sum(m) works just fine without by clause. If we really want to include aggregation into histogram_quantile, it wolud be inconsistent if it would not work without by clause.

WRT topk: topk doesn't even take a by clause. Consistency with topk would be a reason to stick with the original proposal.